x-pack/plugins/file_upload/public/api/index.ts

```diff
@@ -5,16 +5,16 @@
  * 2.0.
  */
 
-import React from 'react';
-import { FileUploadComponentProps, lazyLoadModules } from '../lazy_load_bundle';
+import { lazyLoadModules } from '../lazy_load_bundle';
 import type { IImporter, ImportFactoryOptions } from '../importer';
-import { IndexNameFormProps } from '../';
 import type { HasImportPermission, FindFileStructureResponse } from '../../common';
 import type { getMaxBytes, getMaxBytesFormatted } from '../importer/get_max_bytes';
+import { JsonUploadAndParseAsyncWrapper } from './json_upload_and_parse_async_wrapper';
+import { IndexNameFormAsyncWrapper } from './index_name_form_async_wrapper';
 
 export interface FileUploadStartApi {
   getFileUploadComponent(): ReturnType<typeof getFileUploadComponent>;
```

Comment:
How about just putting the component in the API with no getter function. The function as only needed to wrap the async functionality and now that is contained in the component.

Reply:
see https://github.com/elastic/kibana/pull/100278